sdk/src/main/scala/com/horizen/api/http/SidechainSubmitterApiRoute.scala

```diff
@@ -82,17 +80,17 @@ case class SidechainSubmitterApiRoute(override val settings: RESTApiSettings, ce
     ApiResponseUtil.toResponse(RespSubmitterOk)
   }
 
-  def getSchnorrPublicKeyHash: Route = (post & path("getSchnorrPublicKeyHash")) {
-    entity(as[ReqGetSchnorrPublicKeyHash]) { body =>
-      try {
-        val schnorrPublicKey: SchnorrProposition = new SchnorrProposition(BytesUtils.fromHexString(body.schnorrPublicKey))
-        ApiResponseUtil.toResponse(
-          RespHashSchnorrPublicKey(
-            BytesUtils.toHexString(schnorrPublicKey.getHash)
-          )
-        )
-      } catch {
-        case e: Throwable => SidechainApiError(e)
+  def getKeyRotationMessageToSign: Route = (post & path("getKeyRotationMessageToSign")) {
+    entity(as[ReqGetKeyRotationMessageToSign]) { body =>
+      Try {
+        Await.result(certSubmitterRef ? GetKeyRotationMessageToSign(body.schnorrPublicKey, body.keyType,
```

Comment:
Let's simplify code here.
We don't need to call CertificateSubmitter to get key message to sign. Because Submitter actually doesn't care about it.
Instead you may directly use the following code here
```
if(keyType == 0) {
        CryptoLibProvider.thresholdSignatureCircuitWithKeyRotation.getMsgToSignForSigningKeyUpdate(SchnorrPublicKey.deserialize(schnorrPublicKey.getBytes), withdrawalEpoch,  params.sidechainId)
      } else {
        CryptoLibProvider.thresholdSignatureCircuitWithKeyRotation.getMsgToSignForMasterKeyUpdate(SchnorrPublicKey.deserialize(schnorrPublicKey.getBytes), withdrawalEpoch,  params.sidechainId)
      }
```

Reply:
Done